packages/validator/src/validator.ts

```diff
@@ -136,7 +136,10 @@ export class Validator {
 
         this.clock.runEverySlot(() =>
           this.fetchBeaconHealth()
-            .then((health) => metrics.beaconHealth.set(health))
+            .then((health) => {metrics.beaconHealth.set(health);
+                   if(health == 0){ 
+                    this.logger.warn("Node is unhealthy");
```

Comment:
this can't be checked here as the `getHealth` call [here](https://github.com/ChainSafe/lodestar/blob/8b6ecc44c1892115d0e7396b2a5f0ae8fc0d9e6f/packages/validator/src/validator.ts#L370) will be retried to fallback nodes, meaning this log will only be triggered if all connected nodes are unhealthy which is not what we want as per issue.


Please double check what was noted in the issue, we might have to use the url score tracked in the http client internally to determine if primary node is unhealthy, this also avoids false positives, i.e. a single failed request will not trigger an unhealthy warning.

See also related PR https://github.com/ChainSafe/lodestar/pull/6415 on how we currently track if primary node is unhealthy in metrics.

Reply:
I understood  how to solve it , thanks for the info , will be back with the fix .